pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml

```diff
@@ -562,10 +572,37 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
 
   }]);
 
-  app.controller('SectionProgressController', ['$scope', '$routeParams', '$window', '$q', '$timeout', '$interval', 'sectionsService', 'studentsService',
-                                             function($scope, $routeParams, $window, $q, $timeout, $interval, sectionsService, studentsService) {
+  if (showSectionProjects) {
+    app.controller('SectionProjectsController', ['$scope', '$routeParams', 'sectionsService',
+                                               function($scope, $routeParams,  sectionsService) {
+      $scope.sections = sectionsService.query();
+      $scope.section = sectionsService.get({id: $routeParams.id});
+      $scope.tab = 'projects';
+
+      // Angular does not offer a reliable way to wait for the template to load,
+      // so do it using a custom event here. The call to listen for the custom
+      // event must not be nested inside another deferred call or we might
+      // miss the event.
+      $scope.$on('section-projects-rendered', event => {
+        $scope.section.$promise.then(
+          function (section) {
+            window.renderSectionProjects(section.id);
```

Comment:
FYI, the internet says that using $timeout(fn) or $watch(fn) in the controller should allow the template to be rendered before fn() is called, however this does not appear to be true in practice. I've worked around this by firing a custom event when the template renders.